README.md

```diff
@@ -18,36 +18,203 @@ testing first appeard in Erlang's proprietary QuickCheck. The
 `quickcheck-state-machine` library can be seen as an attempt to provide similar
 functionality to Haskell's QuickCheck library.
 
-### Sample run (teaser)
+### Example
 
-Here's a sample output from when we look for race conditions in the mutable
-reference example:
+Let's implement and test programs using mutable references. Our mutable
+references can be created, read from, written to and incremented:
```

Comment:
Let's mplement and test programs using mutable references implemented using `IORef Int`.